src/consensus/consensus.rs

```diff
@@ -222,10 +234,23 @@ impl Proposer for ShardProposer {
             .as_bytes()
             .to_vec();
 
+        let it = iter::from_fn(|| self.messages_rx.try_recv().ok());
+        let user_messages: Vec<message::Message> = it.collect();
```

Comment:
Does this collect everything until the channel is empty? I think we should impose a max limit and also a timeout (use what's being passed in above).